### DISCUSSION TITLE
Why elasticsearch 5.6?

### DISCUSSION THREAD

**Discussion by FredOloton** [komoot/photon#830] [Q&A]
HI there I was wondering if it was viable and a good idea for me the edit the photon source code locally to use elasticsearch 8.x.x (Not asking for a maintainer or anyone else to do it for me).

Is Elastic 5 still being used due dependency compatibility, remove features or something else that makes it a bad idea to upgrade. I did find that there some discussion that mention it, but none where it is the explicit topic (search queries: elasticsearch upgrade, elasticsearch update, elasticsearch version). 

A part of why I am asking is usual "old version bad, new version good" (vulnerabilities and improvements etc), but we also have custom Elasticsearch analysis and Search that I build for version 8.x.x (So please don't tell me to just use 5.6 as it does make a difference).

I am also aware always just extract the index data and inject it into a new elasticsearch instance with an update veision.
If there's an existing discussion that has all the answers I seek feel free to just answer with an link and close this discussion if appropriate :) .

**Comment by lonvia** [comment]
Please use the experimental version with latest OpenSearch (see #799). It's currently only available on the master branch but there should be a release in the nearish future and eventually ES5.6 support will be phased out and the OpenSearch version will very likely become the only supported version.

For more information on the history of ES5.6 see also #325.

**Reply by FredOloton** [reply]
Thanks for the very quick response I will be reading all of those and look into how I can contribute.
